src/push/utils.ts

```diff
@@ -90,12 +90,51 @@ export function printBytes(bytes: number) {
   return `${bytes.toFixed(1)} ${BYTE_UNITS[exponent]}`;
 }
 
-export function getChunks<T>(arr: Array<T>, size: number): Array<T[]> {
-  const chunks = [];
+export function getBatches(arr: string[], size: number): string[][] {
+  const batches = [];
   for (let i = 0; i < arr.length; i += size) {
-    chunks.push(arr.slice(i, i + size));
+    batches.push(arr.slice(i, i + size));
   }
-  return chunks;
+  return batches;
+}
+
+export function getSizedBatches<T extends { id?: string }>(
+  arr: Array<T>,
+  sizes: Map<string, number>,
+  maxBatchSizeKiB: number,
+  maxBatchItems: number
+): Array<T[]> {
+  const batches: Array<T[]> = [];
+  let currentBatch: T[] = [];
+  let currentSize = 0;
+
+  for (const item of arr) {
+    const sizeKiB = item.id ? sizes.get(item.id) / 1024 : 1;
```

Comment:
The size is stored in bytes, we need to convert them to kibibytes and round it off - https://github.com/elastic/synthetics/pull/993#discussion_r1966379439

Reply:
fixed https://github.com/elastic/synthetics/pull/993/commits/2523967688afbd9f86b867e8070b753b86887054